src/app/EventLogging.h

```diff
@@ -0,0 +1,51 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/ConcreteEventPath.h>
+#include <app/EventLoggingDelegate.h>
+#include <app/data-model/Decode.h>
+#include <app/data-model/List.h> // So we can encode lists
+
+namespace chip {
+namespace app {
+
+template <typename T>
+class EventLogger : EventLoggingDelegate
+{
+public:
+    EventLogger(const T & aEventData) : mEventData(aEventData){};
+    CHIP_ERROR WriteEvent(chip::TLV::TLVWriter & aWriter) final override { return mEventData.Encode(aWriter, TLV::AnonymousTag); }
+
+private:
+    const T & mEventData;
+};
+
+template <typename T>
+CHIP_ERROR LogEvent(const T & aEventData, EndpointId aEndpoint, EventOptions aEventOptions, EventNumber & aEventNumber)
```

Comment:
We need to decide whether we're doing the `EventSchema` thing (which would mean endpoint, etc is included in the EventOptions) or whether we are getting the cluster and event id from the data type and passing in the endpoint like this.

Seems to me that EventSchema should just go away.

That said event priority is not statically deducible from the data type, so we will need to pass that in, via either the EventOptions or some other mechanism.

Reply:
Agreed. 

I'd rather we consolidate around `EventOptions`, remove `EventSchema` (schema is poor term for it anyways) and consequently, remove `mpEventSchema` from `EventOptions`.